python/kserve/kserve/protocol/rest/openai/dataplane.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import AsyncIterator, Dict, Optional, Tuple, Union
+
+from openai.types import Completion, CompletionCreateParams
+from openai.types.chat import ChatCompletion, ChatCompletionChunk
+from openai.types.chat import CompletionCreateParams as ChatCompletionCreateParams
+
+from ...dataplane import DataPlane
+from .openai_model import OpenAIModel
+
+
+class OpenAIDataPlane(DataPlane):
+    """OpenAI DataPlane"""
+
+    async def create_completion(
+        self,
+        model_name: str,
+        request: CompletionCreateParams,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> Union[Completion, AsyncIterator[Completion]]:
+        """Generate the text with the provided text prompt.
+
+        Args:
+            model_name (str): Model name.
+            request (CompletionCreateParams): Params to create a completion.
+            headers: (Optional[Dict[str, str]]): Request headers.
+
+        Returns:
+            response: A non-streaming or streaming completion response.
+
+        Raises:
+            InvalidInput: An error when the body bytes can't be decoded as JSON.
+        """
+        model = self.get_model(model_name)
+        if not isinstance(model, OpenAIModel):
+            raise RuntimeError(f"Model {model_name} does not support completion")
+        return await model.create_completion(request)
```

Comment:
How can the users return custom headers back from the model? It looks like this api will not allow us to do that. This was an issue for v1 endpoints. I feel like this is a good opportunity to fix that issue for the openai endpoints.

Reply:
We have a separate PR to address this 
https://github.com/kserve/kserve/pull/3427

Reply:
That PR seems to be fixing the issue for v1 and v2 endpoints. We want to fix it for openai endpoints too.

Reply:
We need to have the consistent experience, once we finalize the approach for v1/v2 then we can apply to openai endpoints.